game-app/game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -550,41 +543,28 @@ private void setAvailableTech(final Map<String, Map<TechAdvance, Boolean>> value
   }
 
   private Map<String, Map<TechAdvance, Boolean>> getAvailableTech() {
-    return availableTech;
+    return getMapProperty(availableTech);
   }
 
   private void resetAvailableTech() {
     availableTech = null;
   }
 
-  private void setSupport(final String sup) throws GameParseException {
-    if (sup == null) {
-      support = null;
-      return;
-    }
-    final String[] s = splitOnColon(sup);
-    for (int i = 0; i < s.length; i++) {
-      boolean add = true;
-      if (s[i].startsWith("-")) {
-        add = false;
-        s[i] = s[i].substring(1);
-      }
-      boolean found = false;
-      for (final UnitSupportAttachment support :
-          UnitSupportAttachment.get(getData().getUnitTypeList())) {
-        if (support.getName().equals(s[i])) {
-          found = true;
-          if (this.support == null) {
-            this.support = new LinkedHashMap<>();
-          }
-          this.support.put(s[i], add);
-          break;
-        }
-      }
-      if (!found) {
-        throw new GameParseException(
-            "Could not find unitSupportAttachment. name:" + s[i] + thisErrorMsg());
+  private void setSupport(final String value) throws GameParseException {
+    for (final String entry : splitOnColon(value)) {
+      final boolean remove = entry.startsWith("-");
+      final String name = remove ? entry.substring(1) : entry;
+      UnitSupportAttachment.get(getData().getUnitTypeList()).stream()
+          .filter(support -> support.getName().equals(name))
+          .findAny()
```

Comment:
Well that's kinda weird if the value is not actually used.

Reply:
It's meant for validation. Do you prefer noneMatch() and an if statement with the throw in the if?